client/client.go

```diff
@@ -76,13 +77,13 @@ type Client interface {
 	// taking care of region change.
 	// Also it may return nil if PD finds no Region for the key temporarily,
 	// client should retry later.
-	GetRegion(ctx context.Context, key []byte) (*Region, error)
+	GetRegion(ctx context.Context, key []byte, needBuckets bool) (*Region, error)
```

Comment:
Is it a breaking change?

Reply:
Yes. Another way is to add `GetRegionAndBucket()`/`GetPrevRegionAndBucket()`/`GetRegionAndBucketByID()`, but there are many places the `Client` interface is implemented for a mock client, it's also a breaking change and doesn't reduce the workload. Since the main(maybe the only one) user of PD client is client-go and TiDB, I think we can choose either way to utilize the new interface.

Reply:
can we use the option pattern. Or can we put the option into context?

Reply:
OK. I will use the option pattern.